packages/server/src/auth/auth.service.ts

```diff
@@ -28,21 +28,25 @@ export class AuthService {
   }
 
   async createStudentSubscriptions(userId: number): Promise<void> {
-    const memberMailServices = await MailServiceModel.find({
-      where: { mailType: 'member' },
-    });
-if (!memberMailServices) {
-  return
-}
-    const subscriptions = memberMailServices.map((service) => {
-      const subscription = new UserSubscriptionModel();
-      subscription.userId = userId;
-      subscription.serviceId = service.id;
-      subscription.isSubscribed = true;
-      return subscription;
-    });
+    try {
+      const memberMailServices = await MailServiceModel.find({
+        where: { mailType: 'member' },
+      });
+      if (!memberMailServices) {
+        return;
+      }
+      const subscriptions = memberMailServices.map((service) => {
+        const subscription = new UserSubscriptionModel();
+        subscription.userId = userId;
+        subscription.serviceId = service.id;
+        subscription.isSubscribed = true;
+        return subscription;
+      });
 
-    await UserSubscriptionModel.save(subscriptions);
+      await UserSubscriptionModel.save(subscriptions);
+    } catch (err) {
+      console.log(err);
+    }
```

Comment:
This will not inform the user that any error has happened. I was thinking more like a try catch around any call to createStudentSubscriptions that will then return something like "There was a error saving user mail subscriptions"

Reply:
Something like this:
 ```ts
 try {
      this.createStudentSubscription()
    } catch (err) {
      throw new HttpException(
        'There was a error saving user mail subscriptions',
        HttpStatus.INTERNAL_SERVER_ERROR,
      );
    }
```